pynumdiff/finite_difference/_finite_difference.py

```diff
@@ -18,7 +17,7 @@ def first_order(x, dt, params=None, options={}, num_iterations=None):
              - **dxdt_hat** -- estimated derivative of x
     """
     if params != None and 'iterate' in options:
-        warn("""`params` and `options` parameters will be removed in a future version. Use `num_iterations` instead.""")
+        raise DeprecationWarning("""`params` and `options` parameters will be removed in a future version. Use `num_iterations` instead.""")
```

Comment:
I just learned about `DeprecationWarning`, which is more specific to this case and built in, so I'm using that rather than making a separate import.